lib/fungus_toast/games/helpers/cell_grower.ex

```diff
@@ -6,14 +6,16 @@ defmodule FungusToast.CellGrower do
 
   @spec grow_new_cells(integer(), map(), %Player{}) :: [%GridCell{}]
   def grow_new_cells(cell_index, surrounding_cells, player) do
-    # []
-    #   ++ [grow_top_left(surrounding_cells.top_left_cell, player.top_left_growth_chance)]
+    Enum.map(surrounding_cells, fn {k,v} -> try_growing_cell(k, v.index, player) end)
+      |> Enum.filter(fn x -> x != nil end)
   end
 
-  @spec try_growing_cell(integer(), integer(), float()) :: [%GridCell{}]
-  def try_growing_cell(cell_index, player_id, growth_chance) do
+  @spec try_growing_cell(atom(), integer(), %Player{}) :: [%GridCell{}]
+  def try_growing_cell(position, cell_index, player) do
+    {:ok, growth_attribute} = Map.fetch(Player.position_to_attribute_map, position)
+    {:ok, growth_chance} = Map.fetch(player, growth_attribute)
```

Comment:
Per comment below, I feel like we could get rid of line 15 and just do some trickery like `Map.fetch(player, position + "_growth_chance")`. I know that won't work, but I'm just trying to show what I was thinking. Then you only do one Map lookup and the code make a bit more sense

Reply:
I actually originally spent a few minutes trying to do that and I couldn't figure out how since it's an atom. But maybe this just shouldn't be an atom and it would work?

Reply:
You could probably do it as a string and convert to an atom. Using atoms can be dangerous though, since they do not get garbage collected. In this case, it would be fine since there is a finite list of atoms that would be created